### PR TITLE
Link libupnp with pkg_config

### LINKED ISSUE #430: OS X build issue: library not found for -lupnp
I am getting this build error with latest master

``` ssh
   Compiling foxbox v0.1.0 (file:///Volumes/mozilladev/foxbox/foxbox)
error: linking with `cc` failed: exit code: 1
note: "cc" "-m64" "-L" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/foxbox.0.o" "-o" "/Volumes/mozilladev/foxbox/foxbox/target/debug/foxbox" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Volumes/mozilladev/foxbox/foxbox/target/debug" "-L" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps" "-L" "/Volumes/mozilladev/foxbox/foxbox/target/debug/build/openssl-367a67cf98647b2e/out" "-L" "/Volumes/mozilladev/foxbox/foxbox/target/debug/build/openssl-sys-extras-892acb0099c5f8f7/out" "-L" "/usr/local/opt/openssl/lib" "-L" "/Volumes/mozilladev/foxbox/foxbox/target/debug/build/rust-crypto-ee392cda555c7573/out" "-L" "/usr/lib" "-L" "/Volumes/mozilladev/foxbox/foxbox/target/debug/build/openzwave-sys-aa2fac941bf1da6a/out" "-L" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/cargo/git/checkouts/openzwave-rust-3fdf6c635691fa96/master/openzwave-sys/open-zwave" "-L" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib" "-l" "upnp" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libopenzwave_adapter-6e7864586ce16cd1.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libget_if_addrs-04ee712a8f52aa2e.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libfoxbox_users-6e77a44a7e9617ca.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libiron_cors-ab1307d876c26e6e.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libnix-820c8dac842e4b15.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmulticast_dns-546a6fa7308632e7.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libws-19c3c9b728f5620c.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libenv_logger-54fa0d25029b02ef.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libdocopt-bb161c0e33785889.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libjwt-f0321a64b372a700.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmio-2f74fcaa3855d4d7.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libbytes-2fbbdda5bbde5111.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libnix-b32558dd955ccdf5.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmiow-ba6e9462e2153813.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libnet2-559ba40c19f23b6a.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libwinapi-42bc588a308ea664.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libws2_32-2f1486c4a679970f.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libstrsim-7ec45c0f17103caa.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libxml-4a9c7626f62b4c0d.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liburlencoded-6533c5812dfae2bb.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libbodyparser-b483bd4ce8bf70e4.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liburl-8f3a72597d799863.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libtls-258be791819a4d41.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmktemp-9a8e0479ed5ebb2a.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libbitflags-d06009685951cc56.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libip-f5294d306c5e5c51.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/librouter-7b4231f02a8668d7.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libroute_recognizer-d498a0086c418cc4.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libstaticfile-cde1ec256cec3188.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmount-d8b63956a0ae9034.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libslab-22c48cf3d25bac5c.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmount-ab593c7f7630056f.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libsequence_trie-00072ea8be8f33bb.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libregex-a51d4cf4c0db8c6c.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libthread_local-f2c1cfb0d717958d.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libaho_corasick-13668caf93db7fdf.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmemchr-eee90be3d2b5052f.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libthread_id-ed3f7011bd9c5aa2.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libregex_syntax-fca6b3d9bd2fccee.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libsha1-c425125a7e359086.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libbyteorder-8ad3fd7ac7478b9a.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libfoxbox_thinkerbell-f3fe9d9ebf00336e.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libfoxbox_taxonomy-336465ca1a7ef7b6.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libtransformable_channels-201b63da869b7630.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libsublock-8d1766a790a0c5c5.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/librusqlite-174bbbed3cbc60bd.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liblibsqlite3_sys-f35d4b013c8b0194.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libserde_json-6edc82d6f60f8983.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libstring_cache-f778c0f667839a40.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libdebug_unreachable-ec2fe2296ce40c81.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libunreachable-bc14e2c0e4546da4.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libserde-3a777b14c091f0d1.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libphf_shared-d029ae6892b0fb46.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libbitflags-d88a99e5825ec8bf.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libopenzwave_stateful-cf16021e4f4a0aa6.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libopenzwave-838cda4ce99eb110.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libitertools-c7aba225c495f9d3.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libopenzwave_sys-dcbe050cf490846a.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libuuid-5650c0ca096e22f0.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libvoid-e0a8882c9c925513.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libcfg_if-0a8d2242d60f8925.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libkernel32-757937e2863677bb.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libutf8_ranges-a6119bc781af556b.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libtimer-621925c22f6c5c0c.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libchrono-b7e7e442a0c8b7fb.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libpwhash-84a2bacb6eed5026.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libcrypto-bae5fed23a1cbaaf.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liblibc-5eb0f3d71076a4bc.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libc_linked_list-ff0b7ecfb8e6ce0d.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libpersistent-d202c20885f8e9fb.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libiron-f9c7fc83ad20542d.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libhyper-0138d6aa106ddfd5.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liberror-104d955ec44f5893.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libcookie-276cf5002a04c35b.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libopenssl-d6a062adf7371fa2.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libtime-71756e48b8b5b73b.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libtraitobject-3d4dcec5d1662e96.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libbitflags-b378ff20d60f43f8.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libconduit_mime_types-46cd83f432dffe61.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liblanguage_tags-1cb52046c41cf66a.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libunicase-2e75ae83bf996d47.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmodifier-932fdf4731542a6e.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libtypeable-1604229584d39a42.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libsolicit-8f0dfee0deffeb96.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmime-b67d6d6511d55d94.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liblazy_static-5853c3b37be48a6a.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liburl-8ed8652dd12957ed.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libunicode_normalization-f33127ef3e902b05.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libuuid-4661ec327561dea0.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/librand-c724acb3942597d1.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libhttparse-9ed9b694220e1406.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liblazy_static-5e6d9f365bf63baa.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libhpack-320332c60c4dfc72.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liblog-342ffb7444a9471d.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libserde-db1be5227e810a92.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libunicode_bidi-7a56a7dec369a022.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libplugin-444bd670a4e5a6a7.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libtypemap-71d41707b27a594c.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libunsafe_any-558bd65478773c01.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libtraitobject-33ecb530d17570ac.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libnum-98981edb917831dc.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libnum_iter-5766581f7597a504.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libnum_integer-8f7844108d7e0eab.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libnum_traits-14a2d70fdb554c2a.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libopenssl_sys_extras-e04a13dc85856c73.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libopenssl_sys-5e5868bf50eed157.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/librustc_serialize-3bc953984ed46e7f.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libnum_cpus-a5254d7bdb51bb96.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/liblibc-30c6b6751f89189b.rlib" "/Volumes/mozilladev/foxbox/foxbox/target/debug/deps/libmatches-030a774745cc4f96.rlib" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib/libstd-18402db3.rlib" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib/libcollections-18402db3.rlib" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib/librustc_unicode-18402db3.rlib" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib/librand-18402db3.rlib" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib/liballoc-18402db3.rlib" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib/liballoc_jemalloc-18402db3.rlib" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib/liblibc-18402db3.rlib" "/Users/ferjm/.multirust/toolchains/nightly-2016-04-10/lib/rustlib/x86_64-apple-darwin/lib/libcore-18402db3.rlib" "-l" "sqlite3" "-l" "sqlite3" "-l" "sqlite3" "-framework" "IOKit" "-framework" "CoreFoundation" "-l" "c++" "-l" "c" "-l" "m" "-l" "ssl" "-l" "crypto" "-l" "System" "-l" "pthread" "-l" "c" "-l" "m" "-l" "compiler-rt"
note: ld: library not found for -lupnp
clang: error: linker command failed with exit code 1 (use -v to see invocation)

error: aborting due to previous error
error: Could not compile `foxbox`.

To learn more, run the command again with --verbose.
```

I have libupnp installed via brew.

I can workaround this by running `cargo rustc -- -L /usr/local/opt/libupnp/lib` and launching foxbox via `./target/debug/foxbox`. But I cannot use `cargo build` or `cargo run` or any script that uses these commands (i.e. `npm run test-selenium`).

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Fixes #430

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
Thanks to your idea of using pkg_config @julienw, we managed to get Mac OS back running. 

Asking review to the pkg_config master r? @julienw

File: tools/travis-linux-arm_cross_compile.sh [hunk 1]
```diff
@@ -27,27 +30,30 @@ EOF
...
 
     tee -a $HOME/.cargo/config << EOF
-[target.armv7-unknown-linux-gnueabihf]
-linker = "arm-linux-gnueabihf-gcc"
+[target.$RUST_TARGET]
+linker = "$BUILD_TARGET-gcc"
 EOF
+
+    export PKG_CONFIG_LIBDIR="/usr/lib/$BUILD_TARGET/pkgconfig"
+    export PKG_CONFIG_ALLOW_CROSS=1
```
Comment (JohanLorenzo) [inline tools/travis-linux-arm_cross_compile.sh:48]: @fabricedesre This line and the one above will be needed in the Docker image https://hub.docker.com/r/fabricedesre/foxbox-armhf/

File: Cargo.toml [hunk 2]
```diff
@@ -9,11 +9,15 @@ exclude = [ "**/*.conf",
             "**/*.log", "**/*.sqlite", # Application generated data
             "**/*.md", "**/*.yml"      # e.g.: READMEs, Travis
           ]
+links = "upnp"
```
Comment (julienw) [inline Cargo.toml]: please check this works without this line

**Comment by julienw** [comment]
r=me without the "links" line in Cargo.toml. Please wait that travis is green after removing this line. And don't forget to warn @fabricedesre to update the docker build.

**Comment by fabricedesre** [comment]
Not sure if I need any change in the docker image, since it's a "controlled" linux base, but I'll check.

**Comment by julienw** [comment]
> Not sure if I need any change in the docker image, since it's a "controlled" linux base, but I'll check.

I think you'll need it because now we use pkg-config to properly include libupnp. If you don't use these lines pkg-config will fail in a cross-compile environment.

**Comment by JohanLorenzo** [comment]
Links removed and green build (even on Mac OS X). Landing.

**Pull request merged**